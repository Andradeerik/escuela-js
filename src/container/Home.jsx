import React from 'react';
import { connect } from 'react-redux';
import Carousel from '../components/Carousel';
import CarouselItem from '../components/CarouselItem';
import Categories from '../components/Categories';
import FeatureVideo from '../components/FeatureVideo';
import '../assets/styles/Home.scss';

const Home = ({ featureVideo, myList, trends, originals }) => (
  <>
    <FeatureVideo {...featureVideo} />
    <Categories title="Mi lista">
      <Carousel>
        {myList.map(item =>
          <CarouselItem {...item} isList />
        )}
      </Carousel>
    </Categories>
    <Categories title="Tendencias">
      <Carousel>
        {trends.map(item =>
          <CarouselItem key={item.id} {...item} />
        )}
      </Carousel>
    </Categories>
    <Categories title="Originales de Platfix">
      <Carousel>
        {originals.map(item =>
          <CarouselItem key={item.id} {...item} />
        )}
      </Carousel>
    </Categories>
  </>
);

const mapStateToProps = state => {
  return {
    featureVideo: state.featureVideo,
    myList: state.myList,
    trends: state.trends,
    originals: state.originals
  };
};

export default connect(mapStateToProps, null)(Home);
