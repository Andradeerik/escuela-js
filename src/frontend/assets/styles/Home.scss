body {
  margin: 0;
  font-family: $theme-font;
  background-color: $gb-color;
}

* {
  box-sizing: border-box;
}
