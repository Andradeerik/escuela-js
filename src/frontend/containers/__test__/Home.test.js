import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Home from '../Home';

describe('Home component', () => {
  test('Matches the snapshot', () => {
    const home = create(
      <ProviderMock>
        <Home />
      </ProviderMock>,
    );
    expect(home.toJSON()).toMatchSnapshot();
  });
});
