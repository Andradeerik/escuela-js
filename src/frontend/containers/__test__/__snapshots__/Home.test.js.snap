// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Matches the snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Platzi Video"
        className="header__img"
        src="test-file-stub"
      />
    </a>
    <div
      className="header__menu"
    >
      <div
        className="header__menu--profile"
      >
        <img
          alt="Usuario"
          src="test-file-stub"
        />
        <p>
          Perfil
        </p>
      </div>
      <ul>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Iniciar Sesión
          </a>
        </li>
      </ul>
    </div>
  </header>,
  <section
    className="main"
  >
    <h2
      className="main__title"
    >
      ¿Qué quieres ver hoy?
    </h2>
    <input
      className="input isHome"
      placeholder="Buscar..."
      type="text"
    />
  </section>,
  <div
    className="Categories"
  >
    <h3
      className="categories__title"
    >
      Tendencias
    </h3>
    <div
      className="carousel"
    >
      <div
        className="carousel__container"
      >
        <div
          className="carousel-item"
        >
          <img
            alt="In the Dark"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/99118E/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/2"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              In the Dark
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2009 16+ 164
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Instinct"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/302140/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/3"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Instinct
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2002 16+ 137
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Grand Hotel"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/5472FF/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/4"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Grand Hotel
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2014 16+ 163
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Stargate Atlantis"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/B36F20/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/5"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Stargate Atlantis
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2014 16+ 194
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Final Space"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/CCC539/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/6"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Final Space
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2017 16+ 124
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="The InBetween"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/FF7A90/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/7"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              The InBetween
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2011 16+ 179
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="Categories"
  >
    <h3
      className="categories__title"
    >
      Originales de Platfix
    </h3>
    <div
      className="carousel"
    >
      <div
        className="carousel__container"
      >
        <div
          className="carousel-item"
        >
          <img
            alt="Stargate Atlantis"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/306880/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/8"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Stargate Atlantis
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2012 16+ 148
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Alien Highway"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/604180/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/9"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Alien Highway
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2019 16+ 128
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Elementary"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/FF91BA/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/10"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Elementary
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2011 16+ 346
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Strange Angel"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/45807C/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/11"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Strange Angel
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2015 16+ 226
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Private Eyes"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/577380/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/12"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              Private Eyes
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2018 16+ 190
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="NCIS: Los Angeles"
            className="carousel-item__img"
            src="http://dummyimage.com/800x600.png/5472FF/ffffff"
          />
          <div
            className="carousel-item__details"
          >
            <div>
              <a
                href="/player/13"
                onClick={[Function]}
              >
                <img
                  alt="Play Icon"
                  className="carousel-item__details--img"
                  src="test-file-stub"
                />
              </a>
              <img
                alt="Reproducir"
                className="carousel-item__details--img"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <p
              className="carousel-item__details--title"
            >
              NCIS: Los Angeles
            </p>
            <p
              className="carousel-item__details--subtitle"
            >
              2010 16+ 160
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
